querydsl-core/src/main/java/com/querydsl/core/support/CollectionAnyVisitor.java

```diff
@@ -110,7 +110,7 @@ protected Predicate exists(Context c, Predicate condition) {
             Path<?> parent = (Path<?>) expr.getMetadata().getParent().accept(this, context);
             expr = new PathImpl<Object>(expr.getType(), PathMetadataFactory.forCollectionAny(parent));
             EntityPath<?> replacement = new EntityPathBase<Object>(expr.getType(),
-                    ExpressionUtils.createRootVariable(expr));
+                    ExpressionUtils.createRootVariable(expr, replacedCounter++));
```

Comment:
> I think that this particular instance is probably fine.

I don't think discriminating is a good feature in coding style.
But yes, we'll take the conversation there!